include/envoy/upstream/upstream.h

```diff
@@ -110,6 +110,11 @@ class Host : virtual public HostDescription {
    */
   virtual bool healthy() const PURE;
 
+  /**
+   * @return whether in aggregate a host is degraded. A host is never both degraded and healthy.
```

Comment:
My thinking is that we want to be able to degrade a host in two ways: 1) by returning a special header during active health checking and 2) by setting it in EDS. I'll add this to the comment

Reply:
It would end up being a larger change, but if a host can never be both degraded and healthy, it seems like we should convert `healthy()` into an enum? Thoughts on how difficult/scary this would be? I think it's definitely the right way to go.

Reply:
Yeah I think that's probably better, and I don't think it will be that scary, just a bigger change. I'll give that a go and see what it looks like